In [17]:
import numpy as np

### 1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks): zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110], ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]. Используя математические операции, посчитать коэффициенты линейной регрессии, приняв за X заработную плату (то есть, zp - признак), а за y - значения скорингового балла (то есть, ks - целевая переменная). Произвести расчет как с использованием intercept, так и без. 

In [18]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110]) 
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [19]:
X = zp.reshape(10,1)
X

array([[ 35],
       [ 45],
       [190],
       [200],
       [ 40],
       [ 70],
       [ 54],
       [150],
       [120],
       [110]])

In [20]:
Y = ks.reshape(10,1)
Y

array([[401],
       [574],
       [874],
       [919],
       [459],
       [739],
       [653],
       [902],
       [746],
       [832]])

Модель без интерсепта

In [21]:
# Найдем коэффиицент beta
B = np.dot(np.linalg.inv(np.dot(X.T,X)),X.T@Y)
B

array([[5.88982042]])

Ответ: B1=5.88982

Модель с интерсептом

In [22]:
# Добавим единичный столбец к X
X = np.hstack([np.ones((10,1)),X])
X

array([[  1.,  35.],
       [  1.,  45.],
       [  1., 190.],
       [  1., 200.],
       [  1.,  40.],
       [  1.,  70.],
       [  1.,  54.],
       [  1., 150.],
       [  1., 120.],
       [  1., 110.]])

In [23]:
# Найдем коэффииценты beta
B = np.dot(np.linalg.inv(np.dot(X.T,X)),X.T@Y)
B

array([[444.17735732],
       [  2.62053888]])

Ответ: B0=444.17736; B1=2.62054

### 2. В каких случаях для вычисления доверительных интервалов и проверки статистических гипотез используется таблица значений функции Лапласа, а в каких - таблица критических точек распределения Стьюдента?

Ответ: Таблица значений функции Лапласа используется в случаях, когда дисперсия генеральной совокупности известна. Таблица критическимх распределений Стьюдента используется, когда дисперсия генеральной совокупности неизвестна.